File: cpp/to_camel_case.cpp
std::string to_camel_case(std::string text) {
  std::set<char> delimiters = {'-', '_'};
  std::string result = "";
  for (int i=0; i < int(text.size()); ++i){
    if(delimiters.find(text[i]) != delimiters.end()){
      result = result + std::string(1, toupper(text[i+1]));
      ++i;
    } else {
      result = result + std::string(1, text[i]);
    }
  }
  return result;
}

File: python/to_camel_case.py
#https://www.codewars.com/kata/517abf86da9663f1d2000003/python

def to_camel_case(text):
    res = ''
    i = 0

    while i < len(text):
        if text[i] == '-' or text[i] == '_':
            res += text[i+1].upper()
            i += 1
        else:
            res += text[i]
        i += 1
        
    return res

#With using text.title for each word
def to_camel_case_title(text):
    return text[:1] + text.title()[1:].replace('_', '').replace('-', '')
